src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { Select, Store } from '@ngxs/store';
+import { distinctUntilChanged, filter, map, Observable, Subject, takeUntil } from 'rxjs';
+import { Institution, InstituitionHierarchy, InstitutionFieldDescription } from '../../../../../shared/models/institution.model';
+import { GetFieldDescriptionByInstitutionId, GetAllInstitutionsHierarchy } from '../../../../../shared/store/meta-data.actions';
+import { MetaDataState } from '../../../../../shared/store/meta-data.state';
+
+
+@Component({
+  selector: 'app-directions-institution-hierarchies-list',
+  templateUrl: './directions-institution-hierarchies-list.component.html',
+  styleUrls: ['./directions-institution-hierarchies-list.component.scss']
+})
+export class DirectionsInstitutionHierarchiesListComponent implements OnInit, OnDestroy {
+  @Input() institution: Institution;
+
+  @Select(MetaDataState.instituitionsHierarchyAll)
+  institutionsHierarchies$: Observable<InstituitionHierarchy[]>;
+  @Select(MetaDataState.institutionFieldDesc)
+  institutionFieldDesc$: Observable<InstitutionFieldDescription[]>;
+
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  displayedColumns: string[];
+  institutionalHierarchies: InstituitionHierarchy[];
+  records: string[][];
+  isLoaded: boolean = false;
+  dataSource: MatTableDataSource<object> = new MatTableDataSource([{}]);
+
+  constructor(private store: Store) {
+  }
+
+  ngOnInit(): void {
+    this.store.dispatch([new GetFieldDescriptionByInstitutionId(this.institution.id),
+      new GetAllInstitutionsHierarchy()]);
+
+    this.institutionFieldDesc$.pipe(
```

Comment:
How about create separate method for all that stuff ( I mean it is not 7 line of code nad a little bit too much for ngoninit)